### PR TITLE
[NFC][MLIR][TableGen] Eliminate `llvm::` for commonly used types

### PR DESCRIPTION
**Pull request by jurahul**
Eliminate `llvm::` namespace qualifier for commonly used types in MLIR TableGen backends to reduce code clutter.

### CODE REVIEW

File: mlir/tools/mlir-tblgen/AttrOrTypeDefGen.cpp [hunk 1]
```diff
@@ -707,7 +709,7 @@ class DefGenerator {
   void emitParsePrintDispatch(ArrayRef<AttrOrTypeDef> defs);
 
   /// The set of def records to emit.
-  std::vector<const llvm::Record *> defRecords;
+  std::vector<const Record *> defRecords;
```
Comment (kuhar) [inline mlir/tools/mlir-tblgen/AttrOrTypeDefGen.cpp:712]: BTW, why do we use `std::vector` here instead of `SmallVector`. Is this intentional?

**Review by kuhar — APPROVED** [review]
Seems reasonable

File: mlir/tools/mlir-tblgen/AttrOrTypeDefGen.cpp (hunk 1 above)
Comment (jurahul) [reply mlir/tools/mlir-tblgen/AttrOrTypeDefGen.cpp:712]: I am not really sure.

Comment (River707) [reply mlir/tools/mlir-tblgen/AttrOrTypeDefGen.cpp:712]: There isn't a specific reason why to use std::vector over SmallVector here.

File: mlir/tools/mlir-tblgen/FormatGen.cpp [hunk 2]
```diff
@@ -13,6 +13,8 @@
 
 using namespace mlir;
 using namespace mlir::tblgen;
+using llvm::SourceMgr;
+namespace cl = llvm::cl;
```
Comment (River707) [inline mlir/tools/mlir-tblgen/FormatGen.cpp:17]: We generally don't add usings for llvm::cl, I'd worry this is going to add inconsistencies in the tools.

File: mlir/tools/mlir-tblgen/LLVMIRIntrinsicGen.cpp [hunk 3]
```diff
@@ -24,44 +24,51 @@
 #include "llvm/TableGen/Record.h"
 #include "llvm/TableGen/TableGenBackend.h"
 
-static llvm::cl::OptionCategory intrinsicGenCat("Intrinsics Generator Options");
+using llvm::Record;
+using llvm::RecordKeeper;
+using llvm::Regex;
+using llvm::SmallVector;
+using llvm::StringRef;
```
Comment (River707) [inline mlir/tools/mlir-tblgen/LLVMIRIntrinsicGen.cpp:31]: Could you just add a `using namespace mlir`? That should cover the basic ones.

File: mlir/tools/mlir-tblgen/OpInterfacesGen.cpp [hunk 4]
```diff
@@ -23,6 +23,9 @@
 #include "llvm/TableGen/TableGenBackend.h"
 
 using namespace mlir;
+using llvm::interleaveComma;
```
Comment (River707) [inline mlir/tools/mlir-tblgen/OpInterfacesGen.cpp:26]: I'd drop this one, we generally don't add usings for things like this.

**Review by River707 — APPROVED** [review]
Thanks!

File: mlir/tools/mlir-tblgen/FormatGen.cpp (hunk 2 above)
Comment (jurahul) [reply mlir/tools/mlir-tblgen/FormatGen.cpp]: Done, I reverted the cl part of the change

File: mlir/tools/mlir-tblgen/OpInterfacesGen.cpp (hunk 4 above)
Comment (jurahul) [reply mlir/tools/mlir-tblgen/OpInterfacesGen.cpp]: Done

File: mlir/tools/mlir-tblgen/LLVMIRIntrinsicGen.cpp (hunk 3 above)
Comment (jurahul) [reply mlir/tools/mlir-tblgen/LLVMIRIntrinsicGen.cpp]: Done

**Pull request merged**